docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,111 +1,340 @@
 
-    iOSTarget("ios")
+    android()
+    iosArm64()
+    iosSimulatorArm64()
 }
 ```
-{initial-collapse-state="collapsed"}
 
-```bash
-# workaround 2: make symbolic links to use one source set for two targets
-ln -s iosMain iosArm64Main && ln -s iosMain iosX64Main
+When you declare final targets `android`, `iosArm64`, and `iosSimulatorArm64` in your code, the Kotlin Gradle plugin finds
+suitable shared source sets from the template and creates them for you. The resulting hierarchy looks like this:
+
+![An example of using the default target hierarchy](default-hierarchy-example.svg){thumbnail="true" width="350" thumbnail-same-file="true"}
+
+Green source sets are actually created and present in the project, while gray ones from the default template are
+ignored. As you can see, The Kotlin Gradle plugin hasn't created the `watchos` source set, for example, because there
+are no watchOS targets in the project.
+
+If you add a watchOS target, like `watchosArm64`, the `watchos` source set is created, and the code
+from `apple`, `native`, and `common` source sets is compiled to `watchosArm64` as well.
+
+> In this example, `apple` and `native` source sets compile only to `iosArm64` and `iosSimulatorArm64` targets.
+> Therefore, despite their names, they have access to the full iOS API.
+> This might be counter-intuitive for source sets like `native`, as you may expect that only APIs available on all
+> native targets are accessible in this source set. This behavior may change in the future.
+>
+{type="note"}
+
+### Adjust resulting hierarchy
+
+If necessary, you can further configure the resulting hierarchy manually [using the `dependsOn` relation](#manual-configuration).
+To do that, apply the `by getting` construction for the source sets created with `targetHierarchy.default()`.
+
+Consider this example of a project with a source set shared between the `jvm` and `native` targets only:
+
+```kotlin
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
+kotlin {
+    // Enable the default target hierarchy:
+    targetHierarchy.default()
+
+    jvm()
+    iosArm64()
+    // the rest of the necessary targets...
+
+    sourceSets {
+        val commonMain by getting
+
+        val jvmAndNativeMain by creating {
+            dependsOn(commonMain)
+        }
+
+        val nativeMain by getting {
+            dependsOn(jvmAndNativeMain)
+        }
+
+        val jvmMain by getting {
+            dependsOn(jvmAndNativeMain)
+        }
+    }
+}
 ```
-{initial-collapse-state="collapsed"}
 
-Instead of doing this, you can create a hierarchical structure
-with [target shortcuts](multiplatform-share-on-platforms.md#use-target-shortcuts)
-available for typical multi-target scenarios, or you can manually declare and connect the source sets. For example, you
-can create two iOS targets and a shared source set with the `ios()` shortcut:
+Removing `dependsOn` relations automatically created by `targetHierarchy.default()` could be cumbersome.
```

Comment:
It could be cumbersome (taking time) to remove "dependsOn", that was automatically created by the `...()) call.

Or any other way, just to make the sentence more easy to read.